### PR TITLE
Add pushstate functionality and also set the cookies properly

### PR DESCRIPTION
**Pull request by jshoe**
- Redone to use the axios async stuff so it's cleaner.
- Went with adding the background info to the "report_info" call from samples_controller because it seemed cleaner rather than worry about background name and background id not matching in the JS.

### CODE REVIEW

File: app/assets/src/components/PipelineSampleReads.jsx [hunk 1]
```diff
@@ -349,6 +347,61 @@ class PipelineSampleReads extends React.Component {
...
+      // Select the background with the matching name.
+      let match = this.allBackgrounds.filter(
+        b => b["name"] === Cookies.get("background_name")
+      );
+      if (match && match[0] && match[0]["id"]) {
+        let cookie_id = match[0]["id"];
+        let loaded_id = reportPageParams.background_id;
+        if (loaded_id !== cookie_id) {
+          // Refresh the page using this background id.
+          this.refreshPage({ background_id: cookie_id });
```
Comment (yunfangjuan) [inline app/assets/src/components/PipelineSampleReads.jsx]: this shouldn't happen here actually. this function should be called in componentWillMount before fetchreport data and fetchbackground list

**Comment by jshoe** [comment]
Going to rewrite some parts of this since YF explained how to avoid the page refresh to me..

**Comment by jshoe** [comment]
Updated this one if you want to take a look! Cleaner now

File: app/assets/src/components/PipelineSampleReport.jsx [hunk 2]
```diff
@@ -1205,6 +1211,57 @@ class PipelineSampleReport extends React.Component {
...
+  fetchParams(param) {
+    let url = new URL(window.location);
+    return url.searchParams.get(param);
+  }
+
+  // Select the background with the matching name.
+  getBackgroundIdByName(name) {
+    let match = this.all_backgrounds.filter(b => b["name"] === name);
+    if (match && match[0] && match[0]["id"]) {
+      this.props.reportPageParams.background_id = match[0]["id"];
```
Comment (yunfangjuan) [inline app/assets/src/components/PipelineSampleReport.jsx]: you can probably just return match[0]["id"] 
and do 
const background_id = getBackgroundIdByName(name) ; 
if (background_id) this.props.reportPageParams.background_id = background_id 

so getBackgroundIdByName(name) might be reusable in the future.

**Pull request merged**